### PR TITLE
Add integration with Endevor profiles from Zowe CLI

### PR DESCRIPTION
**Pull request by Alexandru-Dumitru**
This PR adds the following abilities:
- create a Zowe CLI Endevor profile from within Explorer for Endevor
- load Endevor profiles already created from Zowe CLI 
- optional possibility to store username in password in the Profile. Afterwards, the user is not asked for them as long as it uses said profile.
    - Otherwise, user will be prompted for them once at the start of the session (first call to Endevor)  

This PR removes the following abilities:
- integration with Endevor Bridge for Git

### CODE REVIEW

File: src/commands/DeleteConnection.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2019 Broadcom.
```
Comment (ishche) [inline src/commands/DeleteConnection.ts:2]: it should be 2020 already :)

File: tsconfig.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
         ],
         "sourceMap": true,
         "rootDir": "src",
-        "strict": true, /* enable all strict type-checking options */
+        "strict": false, /* enable all strict type-checking options */
```
Comment (ishche) [inline tsconfig.json:25]: that seems scary

File: src/commands/DeleteConnection.ts (hunk 1 above)
Comment (Alexandru-Dumitru) [reply src/commands/DeleteConnection.ts:2]: Gotcha 👍

File: tsconfig.json (hunk 2 above)
Comment (Alexandru-Dumitru) [reply tsconfig.json:25]: I understand and share your concern. The reason behind this is that many components that handle profiles comes from Zowe Explorer which don't have this parameter set. This yields a lot of errors and it's an issue I am planning to work in a future iteration.

File: src/commands/DeleteConnection.ts (hunk 1 above)
Comment (Alexandru-Dumitru) [reply src/commands/DeleteConnection.ts]: fixed in 4fb03cd

File: package.json [hunk 3]
```diff
@@ -42,31 +42,47 @@
             },
             {
                 "command": "endevorexplorer.newHost",
-                "title": "New connection",
+                "title": "Add a New Datasource",
```
Comment (zdmullen) [inline package.json:45]: Hey there. Sorry to beat a dead horse. The Endevor team will (soon, I hope) update their documentation to say "Configuration". So if we can use that throughout, that would be great.

File: package.json [hunk 4]
```diff
@@ -42,31 +42,47 @@
...
+                "title": "Delete Profile",
+                "icon": {
+                    "light": "resources/light/delete.svg",
+                    "dark": "resources/dark/delete.svg"
+                }
+            },
             {
                 "command": "endevorexplorer.updateHost",
-                "title": "Update Host",
+                "title": "Rename Datasource",
```
Comment (zdmullen) [inline package.json:69]: Rename Configuration

File: src/commands/HostDialogs.ts [hunk 5]
```diff
@@ -16,84 +16,159 @@ import { URL } from "url";
...
         }
 
-        const url = await HostDialogs.showUrlInput();
-        if (url === undefined) {
-            return;
-        }
-        const newRepo = new Repository("", url, "", "", "");
+        const profile = connection.getEntity().getProfile();
+        const url = `${profile.protocol}://${profile.host}:${profile.port}`;
+        // TODO: add dataSource here when we import endevor-connection profile
```
Comment (zdmullen) [inline src/commands/HostDialogs.ts:113]: Configuration etc... throughout

File: src/commands/HostDialogs.ts [hunk 6]
```diff
@@ -16,84 +16,159 @@ import { URL } from "url";
...
-        EndevorController.instance.saveRepositories();
-        window.showInformationMessage(`Connection ${oldName} was renamed to ${newName}.`);
     }
 
     private static async showHostNameInput(repo: Repository): Promise<string | undefined> {
         return window.showInputBox({
             ignoreFocusOut: true,
-            placeHolder: "Connection name",
-            prompt: "Enter a custom name for the connection.",
+            placeHolder: "Datasource name",
```
Comment (zdmullen) [inline src/commands/HostDialogs.ts:170]: Configuration name...

File: src/ui/tree/EndevorNodes.ts [hunk 7]
```diff
@@ -532,7 +594,7 @@ export class NewRepositoryNode extends EndevorNode {
         // @ts-ignore
         this.command.id = "endevorexplorer.newHost";
 
-        this.label = "New connection";
+        this.label = "Add a New Datasource";
```
Comment (zdmullen) [inline src/ui/tree/EndevorNodes.ts:597]: Add a New Configuration

Comment (Alexandru-Dumitru) [reply src/ui/tree/EndevorNodes.ts]: done it fb1e037

File: src/commands/HostDialogs.ts (hunk 6 above)
Comment (Alexandru-Dumitru) [reply src/commands/HostDialogs.ts]: done in fb1e037

File: src/commands/HostDialogs.ts (hunk 5 above)
Comment (Alexandru-Dumitru) [reply src/commands/HostDialogs.ts:113]: done in fb1e037

File: package.json (hunk 4 above)
Comment (Alexandru-Dumitru) [reply package.json]: done in fb1e037

File: package.json (hunk 3 above)
Comment (Alexandru-Dumitru) [reply package.json]: done in fb1e037

File: src/commands/DeleteHost.ts [hunk 8]
```diff
@@ -20,10 +20,10 @@ export function deleteHost(arg:any){
     if (arg.contextValue === "repository") {
         const repo: Repository | undefined = arg.getRepository();
         if (repo) {
-            vscode.window.showWarningMessage("Delete connection: " + repo.getName() + "?", "OK").then(message => {
+            vscode.window.showWarningMessage("Delete Datasource: " + repo.getName() + "?", "OK").then(message => {
```
[the remark this answers is not in the document]
Comment (Alexandru-Dumitru) [reply src/commands/DeleteHost.ts]: done in fb1e037

**Pull request merged**